harness/determined/transformers/_hf_callback.py

```diff
@@ -20,32 +38,91 @@ def __init__(
         user_data: Optional[Dict[str, Any]] = None,
     ) -> None:
         super().__init__()
-
         self.core_context = core_context
-
         self.filter_metrics = filter_metrics
         self.user_data = user_data
+
         self.load_last_checkpoint(args)
 
-        self.last_metrics: Dict[str, float] = {"train_step": -1, "eval_step": -1}
-        self.searcher_ops = self.core_context.searcher.operations()
-        self.current_op = next(self.searcher_ops)
-        self.updating_searcher = False
-
-        cluster_info = det.get_cluster_info()
-        assert (
-            cluster_info
-        ), "Could not find `cluster_info`, the HF Callback must be run on a Determined Cluster"
-        searcher_config = cluster_info.trial._config["searcher"]
-        self.searcher_metric = searcher_config["metric"]
-        # Custom searchers have a different config structure which need to be handled differently
-        if searcher_config["name"] == "custom":
-            self.searcher_unit = "batches"
-            self.searcher_max_length = self.current_op.length
+        self.last_train_metrics = -1
+        self.last_eval_metrics = -1
+
+        info = det.get_cluster_info()
+        if not info:
+            raise RuntimeError("det.transformers.DetCallback must be run on a Determined cluster")
+        self.info = info
+
+        self.searcher_metric = None
+        self.time_unit = None
+        if self.info.task_type == "TRIAL":
+            searcher_config = self.info.trial._config["searcher"]
+            self._check_searcher_config(searcher_config, args)
+            self.searcher_metric = searcher_config["metric"]
+            self.time_unit = searcher_config.get("time_unit")
+            # Don't allow filtering of the searcher or time_unit metrics.
+            if self.filter_metrics:
+                self.filter_metrics.append(self.searcher_metric)
+                if self.time_unit:
+                    self.filter_metrics.append(self.time_unit)
+
+    def _check_searcher_config(
+        self, cfg: Dict[str, Any], args: transformers.TrainingArguments
+    ) -> None:
+        if args.max_steps > -1:
+            args_unit = "batches"
+            args_len = args.max_steps
         else:
-            self.searcher_unit = list(searcher_config["max_length"].keys())[0]
-            self.searcher_max_length = list(searcher_config["max_length"].values())[0]
-            self._check_searcher_compatibility(args)
+            args_unit = "epochs"
+            args_len = args.num_train_epochs
+
+        # XXX: confirm that we actually shim legacy adaptive_asha to not have max_length
```

Comment:
we don't. i mean, it happens at the "get" level, not in the expconf itself. asha reads from max_length. looks like this code is reflective of that though, so maybe this comment is outdated.

Reply:
we talked about this.  We should be shimming configs into their latest form when they arrive at the master, to keep the master code as simple as possible.